local/network.go

```diff
@@ -1015,3 +1052,48 @@ func (ln *localNetwork) buildFlags(
 		httpHost:  httpHost,
 	}, nil
 }
+
+// Get AvalancheGo version
+func (ln *localNetwork) getNodeSemVer(nodeConfig node.Config) (string, error) {
+	nodeVersionOutput, err := ln.nodeProcessCreator.GetNodeVersion(nodeConfig)
+	if err != nil {
+		return "", fmt.Errorf(
+			"couldn't get node version with binary %q: %w",
+			nodeConfig.BinaryPath, err,
+		)
+	}
+	re := regexp.MustCompile(`\/([^ ]+)`)
+	matchs := re.FindStringSubmatch(nodeVersionOutput)
+	if len(matchs) != 2 {
+		return "", fmt.Errorf(
+			"invalid version output %q for binary %q: version pattern not found",
+			nodeVersionOutput, nodeConfig.BinaryPath,
+		)
+	}
+	nodeSemVer := "v" + matchs[1]
+	return nodeSemVer, nil
+}
+
+// based on avago version, adapt the given flags to the flag naming and content expected by the version
```

Comment:
This explanation sounds much more complicated than it needs to be IMO. Please try something easier, for example `ensure flags are compatible with the running avalanchego version`

Or something like that.